### PR TITLE
[terra-form-select] - Fix select option width and styling

### PR DESCRIPTION
**Pull request by PK106552**
### Summary

**What was changed:**

1. Changed the --terra-form-select-option-padding variable value to the orion-fusion-theme to match consistent padding so that the option name doesn't break to the next line.
2. Changed onHover style.

**Why it was changed:**

1. The padding variable value in the orion-fusion-theme is bigger than in the other themes and causes the option name to break to the next line in the middle of the word even if the option is the same as current option.
2. The onHover styling for the option looks rather like onFocus (dashed border), focus states cannot be the same as the hover states.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10121 

---

Thank you for contributing to Terra.
@cerner/terra

Befor fix : 

Screenshot 2024-02-07 at 10 11 31 AM
Screenshot 2024-02-07 at 10 13 31 AM

After fix : 

Screenshot 2024-02-07 at 10 10 34 AM
Screenshot 2024-02-07 at 10 13 10 AM

### CODE REVIEW

File: packages/terra-form-select/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Changed
+  * Changed --terra-form-select-option-padding value to the orion-fusion-theme to match consistent padding.
```
Comment (sdadn) [inline packages/terra-form-select/CHANGELOG.md:6]:
```suggestion
  * Changed --terra-form-select-option-padding value to match with the orion-fusion-theme for consistency.
```

File: packages/terra-form-select/src/combobox/Menu.jsx [hunk 2]
```diff
@@ -275,26 +274,6 @@ class Menu extends React.Component {
...
-
-    if (!option.props.disabled) {
-      this.setState({ active: option.props.value });
-    }
-
-    if (option.props.onMouseEnter) {
-      option.props.onMouseEnter(event);
-    }
-  }
-
```
Comment (sdadn) [inline packages/terra-form-select/src/combobox/Menu.jsx:297]: I understand that this is now being handled by CSS but will this cause any loss of functionality?

File: packages/terra-form-select/CHANGELOG.md (hunk 1 above)
Comment (PK106552) [reply packages/terra-form-select/CHANGELOG.md]: Updated - https://github.com/cerner/terra-core/pull/4023/commits/d0008541f863ff69be3c5ed0281eb036fd05d3e3

**Review by adoroshk — APPROVED** [review]
1. Ensured that focus styling is now different from hover, both have background but only focus has dashed border in default, fusion and lowlight themes.
2. Menu items padding fixed in fusion theme.
LGTM  
Screenshot 2024-02-08 at 9 48 22 AM

**Comment by PK106552** [comment]
> 1. Ensured that focus styling is now different from hover, both have background but only focus has dashed border in default, fusion and lowlight themes.
> 2. Menu items padding fixed in fusion theme.
>    LGTM
> 
> Screenshot 2024-02-08 at 9 48 22 AM

Yes, focus styling is now different from hover, we have matched same onHover and onFocus style as other terra components.

**Review by sdadn — APPROVED** [review]

**Comment by rahulkumar8cs** [comment]
+1

**Pull request merged**